exercises/routing/README.md

```diff
@@ -0,0 +1,164 @@
+
+## Step 1
+
+Using the `angular-cli` create a new project called `routing-exercise` with the `--routing` flag:
+
+```
+ng new routing-exercise --routing
+```
+
+
+## Step 2
+
+Use the angular CLI to build the components and service we need by running the following commands:
+
+```
+ng generate component home
+ng generate component user-list
+ng generate component user-detail
+ng generate service users
+```
+
+> The components and service to our ng-module automatically, so they are ready to use.
+
+## Step 3
+
+Add the following snippet to `app.component.css`:
+
+```
+ul {
+  list-style-type: none;
+  margin: 0;
+  padding: 0;
+  overflow: hidden;
+  background-color: #333;
+}
+
+li  {
+ float: left;
+}
+
+li a {
+  display: block;
+  color: white;
+  text-align: center;
+  padding: 14px 16px;
+  text-decoration: none;
+}
+
+li a:hover {
+  background-color: #111;
+}
+```
+
+## Step 4
+
+Add the following snippet to the `app.component.html`
+
+```
+<ul>
+   <li>
+      <a>Dashboard</a>
+    </li>
+    <li>
+      <a>Users</a>
+    </li>
+  </ul>
+```
+
+
+## Step 5
+
+Modify the anchor tags we just added by using the `routerLink` directive so that they point to default route `"/"` and the users route `"/users"`. Then add the `router-outlet` directive in `app.component.html`.
+
+
+## Step 6
+
+Import the `HomeComponent` and `UserListComponent`in `app-routing.module.ts`. Update the routes so that the default route loads the `HomeComponent` and the `"/users"` route loads the `UserListComponent`.
+
+## Step 7
+
+Update `users.service.ts` with the following snippet:
+
+```
+export class UsersService {
+
+  users = [{
+    id: "2eifj0s",
+    name: 'Iva Gamble',
+    company: 'FLOTONIC',
+    email: 'iva@flotonic.com'
+  }, {
+    id: "sj0dfj",
+    name: 'Johns Pearson',
+    company: 'Hyplex',
+    email: 'johns@hyplex.me'
+  }, {
+    id: "vn0ekj",
+    name: 'Jane Simmons',
+    company: 'Geoform',
+    email: 'Jane@geoform.me'
+  }, {
+    id: "f02jfi",
+    name: 'John Doe',
+    company: 'Flotonic',
+    email: 'john@flotonic.com'
+  }]
```

Comment:
I think it's better to move this data to its own JSON file called `users.json`. We can provide that file as part of the exercise like we did in the [structural directive exercise](https://github.com/rangle/angular-training-slides/tree/master/exercises/structural-directives)